contracts/test/LightClient.t.sol

```diff
@@ -112,6 +142,124 @@ contract LightClient_constructor_Test is LightClientCommonTest {
+        (LC.LightClientState[] memory states, V.PlonkProof[] memory proofs) =
+            abi.decode(result, (LC.LightClientState[], V.PlonkProof[]));
+
+        newState = states[0];
+        newProof = proofs[0];
+    }
+
+    function test_NoProverPermissionsRequired() external {
+        //set permissioned flag to false
+        vm.expectEmit(true, true, true, true);
+        emit LC.PermissionedProverModeDisabled();
+        vm.prank(admin);
+        lc.setPermissionedProverMode(false);
+
+        //any prover can call the newFinalizedState method as the contract is not in permissioned
+        // prover mode
+        vm.prank(makeAddr("randomUser"));
+        lc.newFinalizedState(newState, newProof);
+    }
+
+    function test_NoProverPermissionsRequiredAndProverNotSet() external {
+        //set permissioned flag to false
+        vm.expectEmit(true, true, true, true);
+        emit LC.PermissionedProverModeDisabled();
+        vm.prank(admin);
+        lc.setPermissionedProverMode(false);
+
+        //assert that the contract is not permissioned
+        assert(lc.permissionedProverMode() == false);
+
+        //any prover can call the newFinalizedState method as the contract is not in permissioned
+        // prover mode
+        vm.prank(makeAddr("randomUser"));
+        lc.newFinalizedState(newState, newProof);
+    }
+
+    function test_NoProverPermissionsRequiredAndNoProverSet() external {
+        //deploy a fresh light client proxy
+        (address newLcTestProxy,,) = deployer.deployContract(genesis, BLOCKS_PER_EPOCH_TEST, admin);
+
+        //cast the proxy to be of type light client test
+        LCTest newLc = LCTest(newLcTestProxy);
+
+        //assert that the contract is not permissioned
+        assert(newLc.permissionedProverMode() == false);
+        //assert that the permissionedProver has not been set and thus the zero address
+        assert(newLc.permissionedProver() == address(0));
+
+        //any prover can call the newFinalizedState method as the contract is not in permissioned
+        // prover mode
+        vm.prank(makeAddr("randomUser"));
+        newLc.newFinalizedState(newState, newProof);
+    }
+
+    function test_RevertWhen_NoPermissionedProverSetInPermissionedMode() external {
+        //deploy a fresh light client proxy
+        (address newLcTestProxy,,) = deployer.deployContract(genesis, BLOCKS_PER_EPOCH_TEST, admin);
+
+        //cast the proxy to be of type light client test
+        LCTest newLc = LCTest(newLcTestProxy);
+
+        //set permissioned prover mode to true but do not set the permissionedProver address so that
+        // it's still address(0)
+        vm.expectEmit(true, true, true, true);
+        emit LC.PermissionedProverModeEnabled();
+        vm.prank(admin);
+        newLc.setPermissionedProverMode(true);
+
+        assert(newLc.permissionedProver() == address(0));
+
+        //any prover can call the newFinalizedState method as the contract is not in permissioned
```

Comment:
Update comment

Reply:
in this case the test does indeed check to see whether the contract is in permissioned mode and it's not